pulsar-client/src/main/java/org/apache/pulsar/client/impl/Hash.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl;
+
+public interface Hash {
+    long makeHash(String s);
```

Comment:
Since we're converting later to an `int`, we should just have an it here.

Reply:
@merlimat MurmurHash returns an unsigned int value, so we have to return "as is" for calculating the partition.

Reply:
Sure, but the number is anyway masked with `& Integer.MAX_INT` so the bit sign will be stripped.

Reply:
@merlimat Does this conversion keep the quality of hash?

Reply:
I think this is not good workaround because this may waste the quality of hash.

Reply:
I don't think that it really changes anything because the hash is then modules to the number of partitions, so all the high order bits are anyway discarded.

Reply:
@merlimat IMO, this conversion breaks the uniform distribution of hash. I concern that message routing is biased to the specific partition.

Reply:
@merlimat After second thought, I found you are right! Excuse me!